### PR TITLE
Add dn based output trigger

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

What is says on the can (plus removed dead vtk code as I don't think we'll ever implement/reactivate it).

Still need
- [x] doc
- [x] changelog
- [ ] tests?

## PR Checklist

- [ ] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by BenWibking — DISMISSED** [review]

**Comment by BenWibking** [comment]
I can't tell whether the macOS CI failure is related or not:
```
The following tests FAILED:
	 68 - regression_test:boundary_exchange (Failed)
	 78 - regression_test:poisson (Failed)
```
It looks like some of the output files don't exist, so it might be.

**Comment by pgrete** [comment]
> I can't tell whether the macOS CI failure is related or not:
> 
> ```
> The following tests FAILED:
> 	 68 - regression_test:boundary_exchange (Failed)
> 	 78 - regression_test:poisson (Failed)
> ```
> 
> It looks like some of the output files don't exist, so it might be.

I think the updated handling of ensuring a last time based output is written might be broken (or not behaving as before any more).
I'm investigating.

**Review by Yurlungur — APPROVED** [review]
Assuming we can make tests pass, LGTM. :+1: for deleting dead code.

**Comment by pgrete** [comment]
The failed tests were related to a change in logic for setting `dt` and `tlim` by default as those tests did not include a temporal driver.
I now reverted that logic to the original state (i.e., `dt=0` by default, which allows for the output being triggered).
I also added changelog and doc.
If all tests pass I'm now happy to merge.

File: doc/sphinx/src/outputs.rst [hunk 1]
```diff
@@ -6,10 +6,20 @@ Outputs
 Outputs from Parthenon are controlled via ``<parthenon/output*>`` blocks,
 where ``*`` should be replaced by a unique integer for each block.
 
+The frequency of outputs can be controlled for each block separately
+and can be triggered by either (simulation) time or cycle, i.e.,
+
+- ``dt = 0.1`` means that the output for the block is written every 0.1
+  in simulation time.
+- ``dt = 100`` means that the output for the block is written every 100
```
Comment (BenWibking) [inline doc/sphinx/src/outputs.rst]: I think this should be `dn = 100`.

**Review by BenWibking — CHANGES_REQUESTED** [review]
Typo in docs. Otherwise LGTM

File: doc/sphinx/src/outputs.rst (hunk 1 above)
Comment (Yurlungur) [reply doc/sphinx/src/outputs.rst]: I agree.

**Review by Yurlungur — APPROVED** [review]
Other than the minor docs typo LGTM

File: doc/sphinx/src/outputs.rst (hunk 1 above)
Comment (pgrete) [reply doc/sphinx/src/outputs.rst]: Done.

**Review by BenWibking — APPROVED** [review]

**Pull request merged**